src/encode_like.rs

```diff
@@ -71,6 +72,34 @@ use crate::Encode;
 /// ```
 pub trait EncodeLike<T: Encode = Self>: Sized + Encode {}
 
+/// Reference wrapper that implement encode like any type that is encoded like its inner type.
+///
+/// # Example
+///
+/// ```rust
+/// # use parity_scale_codec::{EncodeLike, Ref};
+/// fn foo<T: EncodeLike<u8>>(t: T) -> T {
+///     store_t(Ref::from(&t)); // Store t using a reference.
```

Comment:
if you rewrite `store_t(&t)` then it would make compilation fail with:
```
failures:

---- src/encode_like.rs - encode_like::Ref (line 79) stdout ----
error[E0277]: the trait bound `&T: parity_scale_codec::EncodeLike<u8>` is not satisfied
 --> src/encode_like.rs:82:5
  |
6 |     store_t(&t); // Store t using a reference.
  |     ^^^^^^^ the trait `parity_scale_codec::EncodeLike<u8>` is not implemented for `&T`
  |
  = help: consider adding a `where &T: parity_scale_codec::EncodeLike<u8>` bound
note: required by `main::store_t`
 --> src/encode_like.rs:86:1
  |
10| fn store_t<T: EncodeLike<u8>>(t: T) {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

```

Reply:
Ahh, because you removed them here:
https://github.com/paritytech/parity-scale-codec/commit/c324f3e7a76dbf35e164a5b8e3ed4e927bdf99c1#diff-b5e31eeb53d19cc6f8e001503ae451ccL48

Hmm, that is shit :/

Reply:
no they havn't been removed, they have been moved to https://github.com/paritytech/parity-scale-codec/pull/161/files#diff-98179c687f4223e9d3bc0699e47ad4ddR314

which is equivalent, the reason of the move is that now it is implemented at the same place as Encode of reference is implemented.

Reply:
the implementation say &T encode like T but not &EncodeLike<T> encode like T that the point